enterprise/cmd/frontend/internal/dotcom/productsubscription/licenses_db.go

```
@@ -262,10 +265,21 @@ func (s dbLicenses) Count(ctx context.Context, opt dbLicensesListOptions) (int,
 	return count, nil
 }
 
+func (s dbLicenses) Revoke(ctx context.Context, id, reason string) error {
+	q := sqlf.Sprintf(
+		"UPDATE product_licenses SET revoked_at = now(), revoke_reason = %s WHERE id = %s",
+		reason,
+		id,
+	)
+	_, err := s.db.ExecContext(ctx, q.Query(sqlf.PostgresBindVar), q.Args()...)
```

Q: Do we need to handle the not found situation here?

---

Not really, since it will result in an error. Only dotcom site admins can call this endpoint, so I want to propagate the error back to them.

---

will it? maybe we can add a test case _just in case_

---

Sorry, only noticed your comment now. Will add a test case in a followup PR.